packages/renderer/src/lib/pod/DeployPodToKube.svelte

```diff
@@ -161,16 +164,21 @@ async function deployToKube() {
   clearInterval(updatePodInterval);
 
   createdRoutes = [];
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
   let servicesToCreate: any[] = [];
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
   let routesToCreate: any[] = [];
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
   let ingressesToCreate: any[] = [];
 
   // if we deploy using services, we need to get rid of .hostPort and generate kubernetes services object
   if (deployUsingServices) {
     // collect all ports
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
     bodyPod.spec?.containers?.forEach((container: any) => {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
       container?.ports?.forEach((port: any) => {
-        let portName = `${bodyPod.metadata.name}-${port.hostPort}`;
+        let portName = `${bodyPod.metadata?.name}-${port.hostPort}`;
```

Comment:
issue: should we skip if metadata.name is undefined rather than using ? everywhere below

for example I'm not sure `${bodyPod.metadata?.name}-${port.hostPort}` producing 'undefined-something' is expected ?

Reply:
Fixed by 8901f95